### DISCUSSION TITLE
Sega CD in MAME using the Genesis Plus Gx

### DISCUSSION THREAD

**Discussion by youtubeeicarlos** [mamedev/discussions#69] [Ideas]
I recently saw a "variant" called MD+ MAME that used MAME together with the Genesis Plus GX, and I was thinking... Why doesn't MAME do this natively?  This would correct the lack of support for Sega CD in CHD, and some errors that MAME cannot resolve in some Sega Genesis/Mega Drive games (errors in themselves that occur due to the emulation not being prepared for some peculiarities of certain games, and in a way, they were recently discovered).  He would be an exceptional assistant, in what MAME lacks, as whoever created the "variant" I mentioned, did everything alone, he was unable to resolve some issues, since his knowledge of how MAME works is limited, so he could not compares to those of you who have been working on the same project for years.

**Comment by rb6502** [comment]
MAME is built on a framework and is very opinionated about coding style, we can't simply import other emulators.  Genesis Plus GX is also under an incompatible license.

That said, as far as I know many Sega CD games work in MAME already.  I just tried Sonic CD and it plays fine, for instance.
